OpenRA.Mods.Cnc/Traits/Buildings/ProductionAirdrop.cs

```diff
@@ -83,7 +83,13 @@ public override bool Produce(Actor self, ActorInfo producee, string productionTy
 					Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", info.ReadyAudio, self.Owner.Faction.InternalName);
 				}));
 
-				actor.QueueActivity(new Fly(actor, Target.FromCell(w, endPos)));
+				// non-VTOLs can't use Fly, as the TakeOff child activity might take longer than leaving the map,
+				// which could lead to aircraft flying another circle to reach endPos before disappearing.
+				if (aircraftInfo.VTOL)
+					actor.QueueActivity(new Fly(actor, Target.FromCell(w, endPos)));
+				else
+					actor.QueueActivity(new FlyOffMap(actor));
```

Comment:
Why can't VTOLs use `FlyOffMap`?

Reply:
Because `FlyOffMap` just directly ticks `FlyTick`, meaning it will ignore the `VTOL` boolean and take off diagonally.
And we can't use child activities in `FlyOffMap` (at least not for horizontal flying), or we lose the ability to exit the activity as soon as we leave the map.
Though I guess I could queue `TakeOff` as child activity there if it's a VTOL.